aerleon/lib/nokiasrl.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2023 Nokia All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Nokia SR Linux yang ACL generator.
+
+More information about the SR Linux ACL model schema: https://yang.srlinux.dev/
+
+"""
+
+import sys
+from typing import Dict, List, Set, Tuple
+
+from aerleon.lib import openconfig
+from aerleon.lib.policy import Term
+
+if sys.version_info < (3, 8):
+    from typing_extensions import TypedDict
+else:
+    from typing import TypedDict
+
+IPPrefix = TypedDict("IPPrefix", {"prefix": str})
+PortRange = TypedDict("PortRange", {"start": int, "end": int})
+Port = TypedDict("Port", {"value": int, "range": PortRange})
+
+Match = TypedDict(
+    "Match",
+    {
+        "fragment": bool,
+        "first-fragment": bool,
+        "protocol": int,
+        "next-header": int,
+        "source-ip": IPPrefix,
+        "destination-ip": IPPrefix,
+        "source-port": Port,
+        "destination-port": Port,
+    },
+)
+Action = TypedDict("Action", {"accept": None, "drop": None})
+ACLEntry = TypedDict(
+    "ACLEntry",
+    {"sequence-id": int, "action": Action, "match": Match},
+)
+aclEntries = TypedDict(
+    "aclEntries", {"ipv4-filter": List[ACLEntry], "ipv6-filter": List[ACLEntry]}
+)
+
+
+class SRLTerm(openconfig.OCTerm):
+    """Creates the term for the SR Linux ACL."""
+
+    ACTION_MAP = {'accept': 'accept', 'deny': 'drop', 'reject': 'drop'}
+
+    def SetAction(self, dict: dict) -> None:
+        action = self.ACTION_MAP[self.term.action[0]]
+        dict['action'] = {action: {}}
+
+    def SetOptions(self, dict: dict) -> None:
+        # Handle various options
+        opts = [str(x) for x in self.term.option]
+        _match = dict['match'] = {}
```

Comment:
Why the double assignment here? You use `dict['match']` in other sections.

Reply:
ok, made consistent